include/reactor-cpp/scheduler.hh

```diff
@@ -27,123 +28,117 @@ namespace reactor {
+  Worker(Scheduler& scheduler, unsigned int identity)
+      : scheduler_{scheduler}
+      , identity_{identity} {}
+  Worker(Worker&& worker); // NOLINT
+  Worker(const Worker& worker) = delete;
 
-  void start_thread() { thread = std::thread(&Worker::work, this); }
-  void join_thread() { thread.join(); }
+  void start_thread() { thread_ = std::thread(&Worker::work, this); }
+  void join_thread() { thread_.join(); }
 
-  static unsigned current_worker_id() { return current_worker->id; }
+  static auto current_worker_id() -> unsigned { return current_worker->identity_; }
 };
 
 class ReadyQueue {
- private:
-  std::vector<Reaction*> queue{};
-  std::atomic<std::ptrdiff_t> size{0};
-  Semaphore sem{0};
-  std::ptrdiff_t waiting_workers{0};
-  const unsigned num_workers;
+private:
+  std::vector<Reaction*> queue_{};
+  std::atomic<std::ptrdiff_t> size_{0};
+  Semaphore sem_{0};
+  std::ptrdiff_t waiting_workers_{0};
+  const unsigned int num_workers_;
 
- public:
-  ReadyQueue(unsigned num_workers) : num_workers(num_workers) {}
+public:
+  explicit ReadyQueue(unsigned num_workers)
+      : num_workers_(num_workers) {}
 
   /**
-   * Retrieve a ready reaction from the queue.
+   * Retrieve a ready reaction from the queue_.
    *
-   * This method may be called concurrently. In case the queue is empty, the
+   * This method may be called concurrently. In case the queue_ is empty, the
    * method blocks and waits until a ready reaction becomes available.
    */
-  Reaction* pop();
+  auto pop() -> Reaction*;
 
   /**
-   * Fill the queue up with ready reactions.
+   * Fill the queue_ up with ready reactions_.
    *
-   * This method assumes that the internal queue is empty. It moves all
-   * reactions from the provided `ready_reactions` vector to the internal queue,
-   * leaving `ready_reactions` empty.
+   * This method assumes that the internal queue_ is empty. It moves all
+   * reactions_ from the provided `ready_reactions` vector to the internal
+   * queue_, leaving `ready_reactions` empty.
    *
    * Note that this method is not thread-safe. The caller needs to ensure that
-   * no other thread will try to read from the queue during this operation.
+   * no other thread will try to read from the queue_ during this operation.
    */
   void fill_up(std::vector<Reaction*>& ready_reactions);
 };
 
-class Scheduler {
- public:
-  using EventMap = std::map<BaseAction*, std::function<void(void)>>;
+using EventMap = std::map<BaseAction*, std::function<void(void)>>;
 
- private:
-  const bool using_workers;
-  LogicalTime _logical_time{};
+class Scheduler { // NOLINT
+private:
+  const bool using_workers_;
+  LogicalTime logical_time_{};
 
-  Environment* _environment;
-  std::vector<Worker> workers{};
+  Environment* environment_;
+  std::vector<Worker> workers_{};
 
-  std::mutex m_schedule;
-  std::unique_lock<std::mutex> schedule_lock{m_schedule, std::defer_lock};
-  std::condition_variable cv_schedule;
+  std::mutex schedule_;
```

Comment:
The name is misleading now. Rename to `lock_schedule_`? Or maybe better `schedule_mutex_`?